### PR TITLE
Allow changing opt-in checkbox position on the checkout page  [MAILPOET-5807]

### PR DESCRIPTION
**Pull request by NeosinneR**
## Description

This PR adds a new setting to change the position of the opt-in checkbox on the checkout page. 

Screenshot 2024-07-31 at 00 54 38

## Code review notes

I realized that it's not clear which checkout page we run tests against (block or shortcode), so I updated tests to explicitly use one or the other or both to increase test coverage. Especially with AutomateWoo, it turned out to be valuable because changing hooks caused the AutomateWoo opt-in checkbox to display, even if it shouldn't. 

## QA notes

1. Test that all opt-in checkbox positions are working as expected. 
2. Test that the opt-in checkbox on old installations remains positioned before terms and conditions.
3. Test that the opt-in checkbox on new installations is positioned after billing info. 
4. Test that the AutomateWoo checkbox is hidden when MailPoet's is displayed. 

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5807]

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [x] 🚫 I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5807]: https://mailpoet.atlassian.net/browse/MAILPOET-5807?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/Migrations/App/Migration_20240730_212419_App.php
Comment (costasovo) [inline mailpoet/lib/Migrations/App/Migration_20240730_212419_App.php]: Perhaps we could skip the version check and just check if the position setting is missing. 

```php
// value is not necessary or is set
if (
 !$this->settings->get(Subscription::OPTIN_ENABLED_SETTING_NAME, false)
 || $this->settings->get(Subscription::OPTIN_POSITION_SETTING_NAME, false)
 ) {
  return;
 }
 
 $this->settings->set(Subscription::OPTIN_POSITION_SETTING_NAME, Hooks::OPTIN_POSITION_BEFORE_TERMS_AND_CONDITIONS);
 ```
It is a tiny detail, but I like to avoid the version check conditions. They are always a bit risky as we may forget to update the version in case the PR merge is postponed. And we are already reading the settings.

**Review by costasovo — COMMENTED** [review]
@NeosinneR Great job!
Thank you for improving the tests! The checkout functionality is very important so it is great we cover all the combinations.

I have just a tiny suggestion.

File: mailpoet/lib/Migrations/App/Migration_20240730_212419_App.php
Comment (NeosinneR) [reply mailpoet/lib/Migrations/App/Migration_20240730_212419_App.php]: Good call; I've updated the migration. It also made me realize, that I forgot to set the default value in Popuplator, so I also added that.

**Review by costasovo — DISMISSED** [review]
@NeosinneR Thank you for the updates!

**Comment by Aschepikov** [comment]
@NeosinneR Hi everything works well but there is a tiny issue, reproducible with chrome and edge
image

**Comment by NeosinneR** [comment]
@Aschepikov fixed in https://github.com/mailpoet/mailpoet/pull/5764/commits/a7cc7f34ed83db6fad7c83308cbedcbea7538470 :) I think it's small enough change not to ask for another code review, so once the build is green, feel free to approve and merge if it's okay by you

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**
